1.develop_vision_models/evaluate_best_vision_models/nbconverted/evaluate_vision_models.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Evaluate Vision Models
+# Here, the best vision models, thus far, are evaluated according to four metrics on both the training and validation datasets:
+# L1 Loss, L2 Loss, PSNR, and SSIM
+
+# In[1]:
+
+
+import copy
+import pathlib
+import random
+import sys
+from collections import defaultdict
+
+import albumentations as A
+import mlflow
+import numpy as np
+import pandas as pd
+import torch
+from torch.utils.data import random_split
+
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
```

Comment:
I agree with the maintenance problem, since python code isn't as stable, as say, bash code. Although, I think many notebooks using python libraries would be in danger in this case. For example, in my IDE this is a snippet I can call, so this danger could also apply to other snippets (duplicate code).

I'll usually use this approach to find the path in most notebooks, since it makes restructuring the repo easier (i.e. I don't have to change paths). I've considered making it a utility function, but then I would need to specify the path to that utility function, in which case, I may as well just specify the root directory path. Another possibility would be to add the utility function to the same directory as the notebook each time. However, this would lead to the same duplicate code problem, since I would need to copy this utility for each notebook I use.

I think the reviewer time is larger for reviewers outside of the Way Lab